src/device-registry/utils/site-activities.js

```diff
@@ -284,58 +260,46 @@ const locationActivityRequestBodies = (req, res) => {
           name: deviceName,
         };
       }
-      logObject("locationActivityBody", locationActivityBody);
+      logObject("siteActivityBody", siteActivityBody);
       logObject("deviceBody", deviceBody);
-      return { locationActivityBody, deviceBody };
+      return { siteActivityBody, deviceBody };
     } else {
       /****incorrect query parameter....... */
-      return res.status(HTTPStatus.BAD_GATEWAY).json({
+      return res.status(HTTPStatus.BAD_REQUEST).json({
         message: "incorrect query parameter",
         success: false,
       });
     }
   } catch (e) {
-    console.log("error" + e);
+    logElement("error", e);
   }
 };
 
-const isDeviceNotRecalled = async (deviceName, tenant) => {
+const isDeviceRecalled = async (name, tenant) => {
   try {
-    const device = await getModelByTenant(
-      tenant.toLowerCase(),
-      "device",
-      DeviceSchema
-    )
-      .find({ name: deviceName })
-      .exec();
     logText("....................");
-    logText("checking isDeviceNotRecalled....");
-    logObject("device is here", device[0]._doc);
-    const isNotRecalled = device[0]._doc.isActive == true ? true : false;
-    logElement("isActive", device[0]._doc.isActive);
-    logElement("isNotRecalled", isNotRecalled);
-    return isNotRecalled;
+    logText("checking isDeviceRecalled....");
+    let device = await getDetail(tenant, name);
+    logObject("device", device);
+    const isRecalled = device[0].isActive == false ? true : false;
```

Comment:
I think a negation operator could be better here

Reply:
Thanks @NoahNsimbe  for the feedback, shall review and act accordingly.

Reply:
@NoahNsimbe , could you shed some light on this? How would we use it here?

Reply:
`line 284`, if a device is not active then it is recalled, I am thinking of substituting `const isRecalled = device[0].isActive == false ? true : false;` with `const isRecalled = !device[0].isActive` since they are mutually exclusive. Correct me if I misunderstood the flow of the code

Reply:
Makes sense, change made. Thanks.